lib/jekyll/jekyll-feed.rb

```diff
@@ -32,7 +32,7 @@ def source_path
 
     def feed_content
       feed = PageWithoutAFile.new(@site, File.dirname(__FILE__), "", path)
-      feed.content = File.read(source_path).gsub(/\s+([{<])/, '\1')
+      feed.content = File.read(source_path).gsub(/(?<!\")\s+([<{])/, '\1')
```

Comment:
There's no way we can remember to always do that going forward, nor could we expect contributors to deal with an odd behavior like that.